### PR TITLE
Added support for updates to data structures and game files.

### PR DESCRIPTION
**Pull request by omagasohe**
Brings stellar object bounds for stations and wormholes current with 0.9.4
Adds support for minables
Adds support for defense fleets and tribute

### CODE REVIEW

**Comment by stillru** [comment]
+1

File: DetailView.cpp [hunk 1]
```diff
@@ -199,10 +211,33 @@ void DetailView::FleetChanged(QTreeWidgetItem *item, int column)
...
+    unsigned row = item->text(2).toInt();
+    if(row == system->Minables().size())
+        system->Minables().push_back({item->text(0), item->text(1).toInt(), item->text(2).toDouble()});
+    else if(item->text(0).isEmpty() && item->text(1).isEmpty())
+        system->Minables().erase(system->Minables().begin() + row);
+    else if(column == 0)
+        system->Minables()[row].type = item->text(0);
+    else if(column == 1)
+        system->Minables()[row].count = item->text(1).toInt();
+    else if(column == 1)
```
Comment (endless-sky) [inline DetailView.cpp]: Should be "== 2", not "== 1".

File: PlanetView.cpp [hunk 2]
```diff
@@ -59,13 +60,33 @@ PlanetView::PlanetView(Map &mapData, QWidget *parent) :
...
     bribe = new QLineEdit(this);
     bribe->setValidator(new QRegExpValidator(QRegExp("0||0?\\.\\d*"), bribe));
     connect(bribe, SIGNAL(editingFinished()), this, SLOT(BribeChanged()));
+
     security = new QLineEdit(this);
     security->setValidator(new QRegExpValidator(QRegExp("0||0?\\.\\d*"), security));
     connect(security, SIGNAL(editingFinished()), this, SLOT(SecurityChanged()));
 
+    tribute = new QLineEdit(this);
+    tribute->setValidator(new QRegExpValidator(QRegExp("-?\\d*\\.?\\d*"), tribute));
```
Comment (endless-sky) [inline PlanetView.cpp]: These and the following values shouldn't be allowed to be negative.

File: PlanetView.cpp [hunk 3]
```diff
@@ -321,7 +369,72 @@ void PlanetView::SecurityChanged()
...
 
+void PlanetView::TributeChanged()
+{
+    if(object && !object->GetPlanet().isEmpty())
+    {
+        double value = numeric_limits<double>::quiet_NaN();
+        if(!tribute->text().isEmpty())
+            value = tribute->text().toDouble();
+        Planet &planet = mapData.Planets()[object->GetPlanet()];
+        if(planet.Tribute() != value)
```
Comment (endless-sky) [inline PlanetView.cpp]: Note: this won't work with NaN values, because any comparison involving NaN always returns false.

**Pull request merged**